### PR TITLE
Add role field branch, Incorporate faculty and role under Find feature

### LINKED ISSUE #5: As a student, I can filter contacts by role
Description: ...  so that I know who to contact

### LINKED ISSUE #61: As a user, I want my contacts to have a role attribute, so that I know what role they are

### PR DESCRIPTION
**Pull request by shurvirarora**
1) Implement role field in each contact within the contact list

- Role field is case insensitive.

- A case insensitive find is more user-friendly because users cannot be expected to type the exact case of the keywords every time.

Let's,
* create a Class called Role and include it in the Person model
* implement automated testing Role class
* include role field in existing automated tests

Fixes #61 

2) Incorporate faculty and role under Find feature

- Find command can now filter out contacts based on faculty and role, in addition to name.

- A case insensitive find is more user-friendly because users cannot be expected to type the exact case of the keywords every time.

Let's,
* modify the NameContainsPredicate class to NameFacultyRoleContainsPredicate
* implement automated testing NameFacultyRoleContainsPredicate class and FindCommand class

Fixes #5 and #51

### CODE REVIEW

File: src/test/data/JsonAddressBookStorageTest/invalidAndValidPersonAddressBook.json [hunk 1]
```diff
@@ -4,12 +4,13 @@
...
     "email": "hans@example.com",
     "faculty": "Computing",
+    "role": "Professor",
     "address": "4th street"
   }, {
     "name": "Person With Invalid Phone Field",
     "phone": "948asdf2424",
     "email": "hans@example.com",
-    "faculty": "Computing",
+    "faculty": "Professor",
```
Comment (eugenechiaay) [inline src/test/data/JsonAddressBookStorageTest/invalidAndValidPersonAddressBook.json:13]: Hmm, do you mean to have Professor under the role field?

File: src/test/java/seedu/address/model/person/RoleTest.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        assertTrue(Role.isValidRole("Professor"));
+        assertTrue(Role.isValidRole("-")); // one character
+        assertTrue(Role.isValidRole("Teaching Assistant")); // long role
+    }
+
+    @Test
+    public void testEqualsSymmetric() {
+        Role x = new Role("Professor"); // equals and hashCode check name field value
+        Role y = new Role("Professor");
+        assertTrue(x.equals(y) && y.equals(x));
```
Comment (eugenechiaay) [inline src/test/java/seedu/address/model/person/RoleTest.java:41]: We could separate out each assert true into 4 separate assert equals, in that case any failure in the could would be more easily identified

File: src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java [hunk 3]
```diff
@@ -45,25 +48,24 @@ public void toModelType_validPersonDetails_returnsPerson() throws Exception {
     @Test
     public void toModelType_invalidName_throwsIllegalValueException() {
         JsonAdaptedPerson person =
-                new JsonAdaptedPerson(INVALID_NAME, VALID_PHONE, VALID_EMAIL, VALID_FACULTY,
-                        VALID_ADDRESS, VALID_FAVOURITE, VALID_TAGS);
+                new JsonAdaptedPerson(INVALID_NAME, VALID_PHONE, VALID_EMAIL, VALID_FACULTY, VALID_ROLE, VALID_ADDRESS,
```
Comment (eugenechiaay) [inline src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java:51]: Are we taking out address in this commit? If not its g :)

**Review by eugenechiaay — APPROVED** [review]
LGTM

File: src/test/java/seedu/address/model/person/RoleTest.java (hunk 2 above)
Comment (shurvirarora) [reply src/test/java/seedu/address/model/person/RoleTest.java]: Noted on the input! 👍

**Pull request merged**